tests/integration/WireServerTest.cpp

```diff
@@ -121,3 +145,50 @@ TEST_F(SocketServerTest, receiveAndSendsSingleLineMassages) {
     EXPECT_THAT(client, EventuallyReceives("B"));
     EXPECT_THAT(client, EventuallyReceives("C"));
 }
+
+#if defined(BOOST_ASIO_HAS_LOCAL_SOCKETS)
+class UnixSocketServerTest : public SocketServerTest {
+protected:
+    UnixSocketServer *server;
+    UnixSocketServerTest() :
+        server(NULL) {
+    }
+
+    virtual UnixSocketServer* createListeningServer() {
+        fs::path socket = fs::temp_directory_path() / fs::unique_path();
+        server = new UnixSocketServer(&protocolHandler);
+        server->listen(socket.string());
+        return server;
+    }
+
+    virtual void TearDown() {
+        SocketServerTest::TearDown(); // MUST be done _before_ server deletion
+        delete server;
+        server = NULL;
+    }
+};
+
+TEST_F(UnixSocketServerTest, clientCanConnect) {
+    // given
+    stream_protocol::iostream client;
+
+    // when
+    client.connect(server->listenEndpoint());
+
+    // then
+    EXPECT_THAT(client, IsConnected());
+}
+
+TEST_F(UnixSocketServerTest, socketIsRemovedByDestructor) {
+    // given
+    stream_protocol::endpoint socketName = server->listenEndpoint();
+    ASSERT_TRUE(fs::exists(socketName.path()));
+
+    // when
+    (void)stream_protocol::iostream(server->listenEndpoint()); // connect and disconnect
+    UnixSocketServerTest::TearDown();
```

Comment:
Ah yes, I already felt the ordering stuff placed in the inherited `TearDown` was nasty. This is nicer.